skypalwarmup/JSUserRegistrationService/src/test/java/application/controller/UserControllerTest.java

```diff
@@ -0,0 +1,112 @@
+package application.controller;
+
+import application.dao.UserRepository;
+import application.entity.AccountInformation;
+import application.entity.Contacts;
+import application.entity.PersonalDetails;
+import application.entity.User;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.data.domain.Example;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+/**
+ * Copyright (c) 2016-2018, Jumia.
+ */
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class UserControllerTest {
+
+    @Autowired
+    private WebApplicationContext wac;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    private MockMvc mockMvc;
+
+    private User userUnderTest;
+
+    @Before
+    public void setUp() {
+        this.mockMvc = MockMvcBuilders.webAppContextSetup(this.wac).build();
+        userRepository.deleteAll();
+        createUserUnderTest();
+    }
+
+    @After
+    public void tearDown() {
+        userRepository.deleteAll();
+    }
+
+    private void createUserUnderTest() {
+        AccountInformation inTestAccountInformation = new AccountInformation();
+        inTestAccountInformation.setUserName("sho9meUae");
+        inTestAccountInformation.setPassword("somePas$sw1ord");
+
+        Contacts inTestContacts = new Contacts();
+        inTestContacts.setEmail("safgdgom8hgejEfmgailm@ail.com");
+        inTestContacts.setCellPhone("+000 000000");
+
+        PersonalDetails inTestPersonalDetails = new PersonalDetails();
+        inTestPersonalDetails.setAge("30");
+        inTestPersonalDetails.setFirstName("someNjame");
+        inTestPersonalDetails.setLastName("someName");
+
+        userUnderTest = new User();
+        userUnderTest.setAccountInformation(inTestAccountInformation);
+        userUnderTest.setContacts(inTestContacts);
+        userUnderTest.setPersonalDetails(inTestPersonalDetails);
+    }
+
+    private String getUserValueAsJSONString() throws Exception {
+        ObjectMapper om = new ObjectMapper();
+        return om.writeValueAsString(userUnderTest);
+    }
+
+    @Test
+    public void testRegisterCREATED() throws Exception {
```

Comment:
Line 81: For future reference our test naming convention follows something like:

`methodName_whenSomething_returnsSomething` OR `methodName_withSomething_thenSomething`. In this case, we would have:

`register_whenUserDoesNotExist_returnsUserCreated`